frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/ArchiveExtractor.java

```diff
@@ -139,4 +141,26 @@ public void extract(String archive, String destinationDirectory) throws ArchiveE
                     + "'", e);
         }
     }
+
+    /**
+     * Do multiple file system checks that should enable the plugin to work on any file system
+     * whether or not it's case sensitive or not.
+     *
+     * @param destPath
+     * @param destDir
+     * @return
+     */
+    private boolean startsWithPath(String destPath, String destDir) {
+        if (destPath.startsWith(destDir)) {
+            return true;
+        } else if (destDir.length() > destPath.length()) {
+            return false;
+        } else {
+            if (new File(destPath).exists() && !(new File(destPath.toLowerCase()).exists())) {
```

Comment:
I don't understand why this check is here, why is it checking whether a file exists?

Reply:
The path that Maven was feeding your plugin was not the same case as what was on the file system. However, HPFS+ and NTFS are case insensitive by default so the folder does actually exist and technically that exception should not be thrown.

FWIW, we fixed **our** issue by finally stomping the inconsistencies on a Linux box IIRC. Based on the other users' comments, I'm not entirely sure their issues are the same so you might want to just be aware of this particular scenario and ask them for more details.

Reply:
That said, I think all of the other users were running Windows or macOS.